data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -282,37 +294,75 @@ public void doOutput(final Collection<Record<Event>> records) {
 
       BulkOperation bulkOperation;
 
-      if (StringUtils.equalsIgnoreCase(action, BulkAction.CREATE.toString())) {
-
-        final CreateOperation.Builder<Object> createOperationBuilder = new CreateOperation.Builder<>()
-                .index(indexName)
-                .document(document);
-
-        docId.ifPresent(createOperationBuilder::id);
-        routing.ifPresent(createOperationBuilder::routing);
-
-        bulkOperation = new BulkOperation.Builder()
-                .create(createOperationBuilder.build())
-                .build();
+      String eventAction = action;
+      if (actions != null) {
+        for (final Map<String, Object> actionEntry: actions) {
+            final String condition = (String)actionEntry.get("when");
+            eventAction = (String)actionEntry.get("type");
+            if (condition != null &&
+                expressionEvaluator.evaluateConditional(condition, event)) {
+                    break;
+            }
+        }
+      }
+      if (eventAction.contains("${")) {
+          eventAction = event.formatString(eventAction, expressionEvaluator);
+      }
+      if (BulkAction.fromOptionValue(eventAction) == null) {
+        LOG.error("Unknown action {}, skipping the event", eventAction);
+        invalidActionErrorsCounter.increment();
+        continue;
+      }
 
+      if (StringUtils.equalsIgnoreCase(eventAction, BulkAction.CREATE.toString())) {
```

Comment:
Let's require the exact case. Ignoring the case could lead to users errors and inconsistencies. It's also too easy for us to accidentally break support for variable cased input.

Reply:
OK. It is how the existing code is. Hopefully, it won't break if I make it exact case match.

Reply:
OK. If that is the existing behavior, we can keep it for now. Maybe in 3.0 we can consolidate all to use only correct case.